test/test_cpp_extensions.py

```diff
@@ -623,17 +626,133 @@ def test_set_default_type_also_changes_aten_default_type(self):
+                                                             shuffle_chunks=False,
+                                                             shuffle_samples=False)
+
+        trainset = chunk.ChunkDatasetWrapper(dummy_chunkdataset)
+        trainset.reset()
+        trainloader = DataLoader(dataset=trainset,
+                                 batch_size=None,
+                                 pin_memory=True)
+        for i, actual in enumerate(trainloader, 0):
+            expected = [torch.tensor([j], dtype=torch.long)
+                        for j in list(range(batch_size * i, batch_size * i + batch_size))]
+            self.assertEqual(expected, actual)
+
+    def test_foo_chunkdataset_bindings(self):
+        """
+        This class serves as a more complex example on how to use ChunkDataset API python bindings
+
+        This example uses `FooExampleType` struct as `Example` type, which is a custom type
+        and therefore requires additional code to expose it to Python API.
+        """
+
+        def collate_fn(batch):
+            if batch is not None:
+                # Output is a dictionary
+                dict = {}
+
+                # Utils to allocate shared memory for tensors
+                # We need to know the exact amount of memory to preallocate it
+                # in shared memory so the main process can access it quickly
+                # Send each tensor individually to shared memory leads to 2x perf hit
+                features = []
+                features_numel = 0
+                features_out = None
+                labels = []
+                labels_numel = 0
+                labels_out = None
+
+                for u in batch:
+                    # C++ FooExampleType to Pytorch Tensors (zero-copy)
+                    feature_tensor = torch.from_numpy(numpy.array(u.feature_))
+                    label_tensor = torch.from_numpy(numpy.array(u.label_))
+
+                    # Determining tensor size for using shared memory
+                    features_numel += feature_tensor.numel()
+                    features.append(feature_tensor)
+                    labels_numel += label_tensor.numel()
+                    labels.append(label_tensor)
+
+                # Allocating shared memory
+                features_shared_storage = features[0].storage(
+                )._new_shared(features_numel)
+                features_out = features[0].new(features_shared_storage)
+                labels_shared_storage = labels[0].storage()._new_shared(labels_numel)
+                labels_out = labels[0].new(labels_shared_storage)
+
+                # Stacking (copying) tensors into shared memory
+                torch.stack(features, out=features_out)
+                torch.stack(labels, out=labels_out)
+
+                # Return shared memory on torch.multprocessing queues (zero-copy)
+                dict['feature'] = features_out
+                dict['label'] = labels_out
+                return dict
```

Comment:
Does this really have to be this complex? Is that the minimal amount of code you need to test something? What are you trying to check in here?